### PR TITLE
Revised nft full screen view(buy now+sub tabs) issue #1344

### PR DESCRIPTION
**Pull request by Kamran-Khan-Gene**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly reveiw my PR.

**Comment by Kamran-Khan-Gene** [comment]
here are the screenshots.


image


image


image

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my this PR

File: wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart [hunk 1]
```diff
@@ -204,16 +216,8 @@ class _TabFieldState extends State<TabField> {
...
-            style: TextStyle(color: AppColors.kWhite),
-          ),
-          Text(
-            "...",
-            style: TextStyle(color: AppColors.kWhite),
-          ),
-          Text(
-            value.substring(value.length - 5, value.length),
-            style: TextStyle(color: AppColors.kWhite),
+            value.substring(value.length - 10, value.length),
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart:219]: Whats the purpose for -10?

File: wallet/lib/pages/purchase_item/clipper/top_left_bottom_right_clipper.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+class TopLeftBottomRightClipper extends CustomClipper<Path> {
+  @override
+  Path getClip(Size size) {
+    final path = Path();
+
+    path.lineTo(0, 18);
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/purchase_item/clipper/top_left_bottom_right_clipper.dart:8]: Is there any same clipper in the app?

File: wallet/lib/services/data_stores/remote_data_store.dart [hunk 3]
```diff
@@ -126,6 +126,12 @@ abstract class RemoteDataStore {
   /// else will throw error
   Future<List<NftOwnershipHistory>> getNftOwnershipHistory({required String itemId, required String cookBookId});
 
+  /// This method is used to get history of nft owners
+  /// Input: [cookBookID] and [recipeId] of the NFT
```
Comment (kjawadDeveloper2) [inline wallet/lib/services/data_stores/remote_data_store.dart:130]: These fields should match the actual params. `cookBookID` should be `cookbookId`

File: wallet/pubspec.yaml [hunk 4]
```diff
@@ -148,6 +148,10 @@ flutter:
       fonts:
         - asset: assets/fonts/UniversalSans-600.ttf
           weight: 600
+    - family: UniversalSans750
```
Comment (kjawadDeveloper2) [inline wallet/pubspec.yaml:151]: No need to add a different font family add it as a part of UniversalSans.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Kindly add tests.

File: wallet/lib/pages/purchase_item/clipper/top_left_bottom_right_clipper.dart (hunk 2 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/purchase_item/clipper/top_left_bottom_right_clipper.dart:8]: No other clipper was clipping the top right and bottom left corners.

File: wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart (hunk 1 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart:219]: So it was showing 5 character before, 5 characters seems very least.

image


So i added 10 characters here.. 


image

Comment (kjawadDeveloper2) [reply wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart:219]: What if the string is "Ja" what will happen?

Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart:219]: an exception will come.. but these are the addresses and they are long.. if you want me to remove this i will remove it.

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 5]
```diff
@@ -341,7 +298,58 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
...
                       ],
                     ),
-                  )
+                  ),
+                  Column(
+                    children: [
+                      SvgPicture.asset(
+                        SVGUtil.OWNER_VIEWS,
+                        width: 20.w,
+                        height: 15.w,
```
Comment (Amean) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:307]: Should height be .h?

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 6]
```diff
@@ -146,48 +145,49 @@ class _PurchaseItemContentState extends State<PurchaseItemContent> {
...
+        viewModel: viewModel,
+        screen: DetailScreen.purchaseScreen,
+        tapUp: (context) => onTapUp,
+        child: Stack(
+          children: [
+            getTypeWidget(viewModel),
+            Visibility(
+              visible: !viewModel.isViewingFullNft,
+              child: Padding(
+                padding: EdgeInsets.only(left: 8, right: 8, bottom: 8, top: MediaQuery.of(context).viewPadding.top),
```
Comment (Amean) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:160]: Should edgeinsets use screenutil?

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 7]
```diff
@@ -341,7 +298,58 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
...
+                      ),
+                      SizedBox(
+                        width: 4.5.w,
+                      ),
+                      Text(
+                        viewModel.viewsCount.toString(),
+                        style: TextStyle(color: Colors.white, fontSize: 11.sp, fontFamily: kUniversalFontFamily,fontWeight: FontWeight.w700),
+                      ),
+                      SizedBox(
+                        height: 5.w,
```
Comment (Amean) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:317]: Should height be .h?

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 8]
```diff
@@ -341,7 +298,58 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
...
+                      ),
+                      buildLikeColumn(viewModel: viewModel),
+                      SizedBox(
+                        height: 18.w,
```
Comment (Amean) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:321]: Should height be .h?

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 9]
```diff
@@ -419,8 +427,8 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
 
   Widget getLikeIcon({required bool likedByMe}) {
     return SizedBox(
-      height: 20.h,
-      width: 20.h,
+      height: 15.h,
+      width: 15.h,
```
Comment (Amean) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:431]: Should width be .w?

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 10]
```diff
@@ -441,19 +450,21 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
...
               child: Center(
-                  child: IconButton(
-                alignment: Alignment.topRight,
-                padding: const EdgeInsets.only(
-                  bottom: 8,
-                  left: 8,
+                child: IconButton(
+                  key: const Key(kCloseBottomSheetKey),
+                  alignment: Alignment.topRight,
+                  padding: const EdgeInsets.only(
```
Comment (Amean) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:456]: Also use screenutil here?

**Review by kjawadDeveloper2 — APPROVED** [review]

**Pull request merged**